apriltag_detection_node:
  ros__parameters:

    camera_info: "arena_camera/camera_info" #p is used
    image_rect: "arena_camera/image_rect"  #must be rectified

    camera_frame: "arena_camera_optical"

    tag:
      family: "tagStandard41h12"
      ids:    [12, 31, 2]         # tag IDs for which to publish transform
      frames: ["arena_perception/robot_top_tag", "arena_perception/robot_bottom_tag", "arena_tag"]   # frame names
      sizes:  [0.0778, 0.0778, 0.0833]     # tag-specific edge size, overrides the default 'size'

    debug_image: true
    debug_image_topic: "arena_perception/apriltag_detection/debug"


find_camera_in_world_service:
  ros__parameters:
    
    # Frame names
    world_frame: "world"
    calibration_marker_frame: "arena_tag"
    camera_frame: "arena_camera_optical"
    
    # Marker position in world
    marker_position:
      x: 1.275
      y: 0.375
      z: 0.0
    
    # Marker orientation in world (radians)
    marker_orientation:
      roll: 0.0
      pitch: 3.14159
      yaw: 1.5708
    
    # Rates
    calibration_attempt_rate: 2.0
    publish_rate: 60.0
    transform_timeout: 0.1
    median_filter_window: 10  #1 is diabled filtering

    #calibration_file: "package://arena_perception/config/world_to_camera_calibration.temp.yaml"  #optional

robot_detection_node:
  ros__parameters:
    camera_frame: "arena_camera_optical"
    base_frame: "robot/base_footprint"
    detection_base_frame: "arena_perception/robot/base_footprint"
    visible_marker_topic: "arena_perception/robot/visible_marker"
    robot_pose_topic: "arena_perception/robot/pose"
    update_rate: 60.0
    transform_timeout: 0.2
    
    # Simple separate lists
    detection_marker_frames: ["arena_perception/robot_top_tag", "arena_perception/robot_bottom_tag"]
    robot_marker_frames: ["robot/top_apriltag_optical", "robot/bottom_apriltag_optical"]
    marker_names: ["top", "bottom"]